src/couch/src/couch_db_engine.erl

```diff
@@ -891,3 +892,31 @@ finish_compaction(Db, CompactInfo) ->
     end,
     ok = gen_server:call(couch_server, {db_updated, NewDb}, infinity),
     {ok, NewDb}.
+
+
+trigger_on_compact(DbName) ->
+    {ok, DDocs} = get_ddocs(DbName),
```

Comment:
Here we are calling high level API from low level API. Which is a sign of architectural problems. I am ok with that as long as @davisp is aware of the issue.

As an alternative solution we might consider passing ddoc enumerator function (one way or another) down the stack (which I am sure would complicate things).